### ISSUE TITLE
Inconsistent model inference time

### ISSUE THREAD

**Issue by asr-pub** [NVIDIA/NeMo#3123]
Recently, I want to measure the inference time, I design two cases, 
First I read model input from files, 
Second, each model input from the standard input.

While I found first case inference time is about 70 ms except the first call, the second case starts to be highly inconsistent, ranging from 72 ms to 483 ms. 

Why this happened ? How can I make the second case inference time keep consistent ?

```python
model = TokenClassificationModel.restore_from(restore_path=checkpoint_path)
# first case code
for line in open(txt_file, 'r'):
    torch.cuda.synchronize()
    start = time.perf_counter()
    result = model.infer_one_text(text)
    torch.cuda.synchronize()
    elapsed_time = time.perf_counter() - start
    print(f'elapsed_time:{elapsed_time:.6f}', flush=True)

# first case log
elapsed time:0.289745
elapsed time:0.072184
elapsed time:0.060471
elapsed time:0.062084
elapsed time:0.063837
```

```python
# second case code
model = TokenClassificationModel.restore_from(restore_path=checkpoint_path)
while True:
    text = input("text:")
    torch.cuda.synchronize()
    start = time.perf_counter()
    result = model.infer_one_text(text)
    torch.cuda.synchronize()
    end = time.perf_counter() - start
    print(f'elapsed time: {end:.6f}', flush=True)

# second case log
elapsed time:0.483897
elapsed time:0.089466
elapsed time:0.450116
elapsed time:0.072568
elapsed time:0.469203
```

**Comment by okuchaiev** [comment]
why are you calling 
```
torch.cuda.synchronize()
```
Also, I'd double-check that the mode is on GPU (model = model.cuda()) and set it into eval mode (model.eva())

**Comment by asr-pub** [comment]
> why are you calling
> 
> ```
> torch.cuda.synchronize()
> ```
> 
> Also, I'd double-check that the mode is on GPU (model = model.cuda()) and set it into eval mode (model.eva())

Hi okuchaiev, Thank you for replaying.

1. With the topic [Why time.time() in python is inaccurte?](https://discuss.pytorch.org/t/why-time-time-in-python-is-inaccurte/94274/9?u=ken_zheng) , to get the accurate measure time, I use the torch.cuda.synchronize() before and after the inference code.
2. the infer_one_text code is below, that contain `model.cuda()` and `model.eva()`
```python
def infer_one_text(self, text):
    # Switch model to evaluation mode
    device = 'cuda' if torch.cuda.is_available() else 'cpu'
    self.eval()
    self.to(device)
    all_probs = []
    input_ids, input_type_ids, input_mask, subtokens_mask = self.construct_bert_input(text)

    logits = self.forward(
        input_ids=input_ids.to(device),
        token_type_ids=input_type_ids.to(device),
        attention_mask=input_mask.to(device),
    )

    subtokens_mask = subtokens_mask > 0.5
    preds = tensor2list(torch.argmax(logits, axis=-1)[subtokens_mask])
    all_probs.append(preds)

    preds = all_probs[0]
    ids_to_labels = {v: k for k, v in self._cfg.label_ids.items()}
    labels = [ids_to_labels[pred] for pred in preds]
    return labels
```
